src/Resources/ResourceManager/Implementation/ResourceGroups/ExportAzureResourceGroupCmdlet.cs

```diff
@@ -93,51 +95,43 @@ public class ExportAzureResourceGroupCmdlet : ResourceManagerCmdletBaseWithApiVe
         [Parameter(Mandatory = false, HelpMessage = "Do not ask for confirmation.")]
         public SwitchParameter Force { get; set; }
 
+        /// <summary>
+        /// Gets or sets the API version.
+        /// </summary>
+        [CmdletParameterBreakingChange("ApiVersion", ChangeDescription = "Parameter is being deprecated without being replaced. Using the lastest possible API version will become the default behavior.")]
+        [Parameter(Mandatory = false, HelpMessage = "When set, indicates the version of the resource provider API to use. If not specified, the API version is automatically determined as the latest available.")]
+        [ValidateNotNullOrEmpty]
+        public override string ApiVersion { get; set; }
+
         /// <summary>
         /// Executes the cmdlet.
         /// </summary>
         protected override void OnProcessRecord()
         {
             base.OnProcessRecord();
+            string contents;
+
             if (ShouldProcess(ResourceGroupName, VerbsData.Export))
             {
 
                 var resourceGroupId = this.GetResourceGroupId();
 
-                var apiVersion = this.ApiVersion ?? DefaultApiVersion;
-
-                var parameters = new ExportTemplateParameters
-                {
-                    Resources = this.GetResourcesFilter(resourceGroupId: resourceGroupId),
-                    Options = this.GetExportOptions(),
-                };
-
-                var operationResult = this.GetResourcesClient()
-                    .InvokeActionOnResource<JObject>(
-                        resourceId: resourceGroupId,
-                        action: Constants.ExportTemplate,
-                        parameters: parameters.ToJToken(),
-                        apiVersion: apiVersion,
-                        cancellationToken: this.CancellationToken.Value)
-                    .Result;
-
-                var managementUri = this.GetResourcesClient()
-                    .GetResourceManagementRequestUri(
-                        resourceId: resourceGroupId,
-                        apiVersion: apiVersion,
-                        action: Constants.ExportTemplate);
-
-                var activity = string.Format("POST {0}", managementUri.PathAndQuery);
-                var resultString = this.GetLongRunningOperationTracker(activityName: activity,
-                    isResourceCreateOrUpdate: false)
-                    .WaitOnOperation(operationResult: operationResult);
-
-                var template = JToken.FromObject(JObject.Parse(resultString)["template"]);
-
-                if (JObject.Parse(resultString)["error"] != null)
+                if (this.IsParameterBound(c => c.ApiVersion))
```

Comment:
This logic seems like it should be reversed - if we have a apiVersion provided we should go through the Rest API client, otherwise SDK client.